### PR TITLE
DIG-1669: Automate incrementing schema version when schemas get updated

### PR DESCRIPTION
**Pull request by SonQBChau**
## What's new
- The update-schema.yaml action now only triggers on pull requests.
- Removed SHA ref and replaced it with schema_version to avoid circular updates and self-references.
- schema_version.txt is a standalone file.
- Any changes to models or schemas will regenerate the schema and increment the schema version.
- Should not trigger on regular pull requests.

### How to test
- Only code review is needed; this GitHub action only triggers on pull requests, so I will test it out in the next PR.

### CODE REVIEW

**Comment by SonQBChau** [comment]
The number in schema_version.txt is just a random number. It could be 1, but I put 440 because katsu version is 4.4.0.

File: .github/workflows/update-schema.yaml [hunk 1]
```diff
@@ -36,37 +36,28 @@ jobs:
...
-      
+
       - name: Install PyYAML
         run: pip install PyYAML
 
       - name: Convert schema.json to schema.yml
         run: python -c 'import json, yaml; json.load(open("chord_metadata_service/mohpackets/docs/schema.json")); print(yaml.dump(json.load(open("chord_metadata_service/mohpackets/docs/schema.json"))))' > chord_metadata_service/mohpackets/docs/schema.yml
 
-      - name: Update schemas with SHA
+      - name: Increment schema version
```
Comment (daisieh) [inline .github/workflows/update-schema.yaml]: Is there a reason we can't use the SHA of the updated schema.yml (or schema.json, whichever) document as the schema version?

**Review by daisieh — COMMENTED** [review]
I would like the schema version value to be meaningful in some way. Could it be the commit sha of schema.json or schema.yml? Once those have been updated, their shas should stay consistent if we're not overwriting them in a github action, right?

File: .github/workflows/update-schema.yaml (hunk 1 above)
Comment (SonQBChau) [reply .github/workflows/update-schema.yaml]: The problem is the current branch SHA will get deleted once it merges. Sure I can put the SHA in, but if someone were to go back and find it, they won't be able to.

Comment (SonQBChau) [reply .github/workflows/update-schema.yaml]: unless we want to keep all the branchs PR around

Comment (daisieh) [reply .github/workflows/update-schema.yaml]: Not the branch SHA, the file sha. That stays around.

Comment (daisieh) [reply .github/workflows/update-schema.yaml]: For example, in this deleted branch https://github.com/CanDIG/htsget_app/blob/9d00ade5811bf48727b7b73516c30d06796d5f81/tests/test_htsget_server.py and in the stable branch it was merged into https://github.com/CanDIG/htsget_app/blob/stable/tests/test_htsget_server.py, the sha for test_htsget_server.py is 6429207.

Comment (SonQBChau) [reply .github/workflows/update-schema.yaml]: Sure, I can try that

Comment (daisieh) [reply .github/workflows/update-schema.yaml]: maybe that's not the file sha, I mean the commit sha.

Comment (daisieh) [reply .github/workflows/update-schema.yaml]: 6429207 is the sha for the commit, whether it is in the deleted merged PR branch or in the stable branch it was merged into. So if a change was made to schema.json, which is triggering this schema version update, there should be a corresponding commit sha for that.

**Comment by SonQBChau** [comment]
schema_version now uses commit version instead of auto-increasing number

File: chord_metadata_service/mohpackets/utils.py [hunk 2]
```diff
@@ -1,39 +1,17 @@
...
-    except Exception as e:
-        logger.debug(
-            f"An error occurred while fetching the schema URL. Details: {str(e)}"
-        )
-
-    return schema_url
+    file_path = "chord_metadata_service/mohpackets/docs/schema_version.txt"
+    with open(file_path, "r") as file:
+        version = file.read().strip()
+    return int(version)
```
Comment (daisieh) [inline chord_metadata_service/mohpackets/utils.py]: This should be a string, since it's a SHA and not an int.

Comment (daisieh) [inline chord_metadata_service/mohpackets/utils.py]:
```suggestion
    return version
```

**Review by daisieh — APPROVED** [review]
You probably don't want to force the schema_version to an int in get_schema_version(); otherwise this looks good.

File: chord_metadata_service/mohpackets/utils.py (hunk 2 above)
Comment (SonQBChau) [reply chord_metadata_service/mohpackets/utils.py]: good catch!

**Pull request merged**